src/model/value-objects/rundown-event.ts

```diff
@@ -37,16 +38,70 @@ export interface PartSetAsNextEvent extends PartEvent {
   type: RundownEventType.SET_NEXT
 }
 
-export interface RundownAdLibPieceInsertedEvent extends RundownEvent {
-  type: RundownEventType.AD_LIB_PIECE_INSERTED,
-  segmentId: string
-  partId: string
-  adLibPiece: {
-    id: string
-    name: string
-    layer: string
+// TODO: Find a better way to type Parts and Piece for Inserted events.
+export interface PartInsertedAsOnAirEvent extends RundownEvent {
```

Comment:
The type defined for `part` in this and `PartInsertedAsNextEvent` appears to be 100% duplicates. Might not completely solve the TODO, but pulling those to an interface to eliminate the duplication is a good first step. 

That said i don't know of a good name for that interface, which might also be why you created the TODO.

Reply:
It's deliberately implemented as is for now, so we don't try to come up with a half baked solution for this and just stick with.
Anders is going to take a look at this while he implements the frontend part of inserting Parts/Pieces.

Reply:
As long as we agree that the duplication is gone before release, i am happy with it.
Fell free to mark resolved, ones read.

Reply:
The plan is to move these types to the presentation layer and use the DTO definitions for the entity content.

Reply:
@KvelaGorrrrnio But they are used on in the bussiness-logic-layer, which is not allowed to know about the presentation layer. How do you plan to handle that?